src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/VBDefineConstantsEncoding.cs

```diff
@@ -71,28 +83,56 @@ static IEnumerable<string> ReadEntries(string rawText)
                         escaped = false;
                     }
                 }
-                throw new FormatException("Expected valid name value pair.");
+                throw new FormatException("Expected valid name value pair for defining custom constants.");
+            }
+
+            static string SplitSingleEntry(string entry)
+            {
+                bool escaped = false;
+                for (int i = 0; i < entry.Length; i++)
+                {
+                    if (entry[i] == '=' && !escaped)
+                    {
+                        throw new FormatException($"Expected valid name value pair for defining custom constants.");
+                    }
+                    else if (i == entry.Length - 1)
+                    {
+                        return entry;
+                    }
+                    else if (entry[i] == '/')
+                    {
+                        escaped = !escaped;
+                    }
+                    else
+                    {
+                        escaped = false;
+                    }
+                }
+                throw new FormatException("Expected valid name value pair for defining custom constants.");
             }
 
             static string DecodeCharacters(string value)
             {
 
                 return value.Replace("/\"", "\"").Replace("/=", "=").Replace("/,", ",").Replace("//", "/");
             }
```

Comment:
nit: let's remove the blank line in 116.
```suggestion
            static string DecodeCharacters(string value)
            {
                return value.Replace("/\"", "\"").Replace("/=", "=").Replace("/,", ",").Replace("//", "/");
            }
```